### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by mlocher**
* Optimize the Dockerfile a bit (by manually installing the Node.JS repository and not relying on the shell script they provide)
* Resize and minify images when post processing the site. (Max width is 1024px, images are not upscaled if they are smaller)

Staging build available at http://documentation.codeship.com/staging/optimize-images/

### CODE REVIEW

**Comment by mlocher** [comment]
Rebased on current `master` branch. Staging build available at http://documentation.codeship.com/staging/optimize-images/

**Comment by ethangj** [comment]
Everything looks good on staging.

File: gulpfile.js [hunk 1]
```diff
@@ -21,12 +23,42 @@ gulp.task('css', folders(site, function(folder) {
...
+	return gulp.src(paths)
+		.pipe(imageResize({
+			width: 1024,
+			upscale : false
+		}))
+		.pipe(print())
+		.pipe(gulp.dest(path.join(site, 'images')));
+});
+
+gulp.task('img:minify', ['img:resize'], function() {
```
Comment (BoVice) [inline gulpfile.js]: Does `gulp-imagemin` support minify? Would we need to add `gulp-minify` in our deps to support this?

https://www.npmjs.com/package/gulp-minify
https://www.npmjs.com/package/gulp-imagemin

Comment (mlocher) [inline gulpfile.js]: Both tasks are run in https://codeship.com/projects/102044/builds/2c738b0e-a8b9-4a2e-b38d-a469f99b64fe (`post-process` step).

**Comment by BoVice** [comment]
lgtm

**Pull request merged**